common/SC_EventLoop.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <functional>
+#ifdef __APPLE__
+#    include "SC_Apple.hpp"
+#    include <thread>
+#endif
+
+// For now, the event loop is only implemented for macOS where
+// it is most needed (because the GUI event loop *must* run on the main thread).
+// Later a Windows and Linux implementation might follow, but this would
+// require a plugin API method to retrieve a handle to the GUI thread.
+// (Windows and Linux don't have any notion of a "main thread" and therefore
+// lack a method like "dispatch_get_main_queue".)
+
+class EventLoop {
+public:
+    // This function must be called at the very beginning of the main thread!
+    static void setup() {
+#ifdef __APPLE__
+        SC::Apple::EventLoop::setup();
+#endif
+    }
+    // Run the event loop until 'waitFunction' returns.
+    static void run(std::function<void()> waitFunction) {
+        if (!waitFunction) {
+            throw std::runtime_error("EventLoop::run() called without wait function!");
```

Comment:
unnecessarily defensive -- anyone using this function should be very careful about how they use it to begin with.